### PR TITLE
Add sync and publish serializers

### PR DESCRIPTION
**Pull request by asmacdo**
https://pulp.plan.io/issues/3597
fixes #3597

### CODE REVIEW

**Comment by daviddavis** [comment]
We need these exact same serializers in pulp_ansible. Thoughts on moving them to pulpcore?

**Comment by daviddavis** [comment]
A couple things I noticed when looking at the api schema:

* I don't see your changes in the api schema. I think you need to set the serializer class (e.g. `serializer_class= RepositorySyncURLSerializer`) for the `@detail_route` decorator calls.
* Also, since the api schema now has these definitions based on serializer names (e.g. `RepositorySyncURL`) you need to namespace your serializers so they don't conflict with other plugins. Or you can move these serializers to pulpcore.

**Comment by asmacdo** [comment]
For now, I renamed the serializers and kept them. I filed https://pulp.plan.io/issues/3617 to add sync and publish serializers to core.

**Comment by daviddavis** [comment]
That works. Thanks @asmacdo.

**Comment by dralley** [comment]
@asmacdo if we're merging this here, could you go ahead and include this fix?

File: pulp_python/app/serializers.py [hunk 1]
```diff
@@ -203,5 +204,61 @@ class Meta:
...
+        if not repository and not repository_version:
+            raise serializers.ValidationError(
+                _("Either the 'repository' or 'repository_version' need to be specified"))
+        elif not repository and repository_version:
+            return data
+        elif repository and not repository_version:
+            version = core_models.RepositoryVersion.latest(repository)
+            if version:
+                new_data = {'repository_version': version}
+                new_data.update(data)
```
Comment (asmacdo) [inline pulp_python/app/serializers.py]: I'm not a huge fan of altering the data at validation time like this.

**Pull request merged**